### PR TITLE
Some functions that use script triggers were not properly being set as having a repeated capability

### LINKED ISSUE #831: Some functions in nifgen that have repeated capabilities aren't generated properly
nifgen has script triggers and marker events, and the generator doesn't by default catch those inputs as repeated capabilities, so the methods are generated incorrectly.

Example: `configure_digital_level_script_trigger(self, trigger_id, source, trigger_when)`... trigger_id is repeated capability so the usage should be:

``` python
session.ScriptTriggers[0].configure_digital_level_script_trigger(self, source, trigger_when)
```

We didn't catch this in the system test... it is:

src/nifgen/system_tests/test_system_nifgen.py:    
``` python
session.configure_digital_level_script_trigger('ScriptTrigger0', 'PXI_Trig0', nifgen.TriggerWhen.HIGH)
```

This may go away depending what we do about #827

### PR DESCRIPTION
**Pull request by texasaggie97**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [X] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.
- [X] I've ~~added~~fixed tests applicable for this pull request
### What does this Pull Request accomplish?
* Explicitly set `'is_repeated_capability'` for parameters that aren't names with some variant of channel
* Fix nidcpower documentation to refer to public `initiate` rather than `_initiate`.

### List issues fixed by this Pull Request below, if any.
* Fix #831 

### What testing has been done?
* Unit
* System

### CODE REVIEW

File: src/nifgen/metadata/functions_addon.py [hunk 1]
```diff
@@ -84,8 +84,17 @@
     'ConfigureDigitalLevelScriptTrigger':       { 'parameters': { 3: { 'enum': 'TriggerWhen',               }, }, },
 }
 
+# Need to manually set some repeated capabilities
+functions_rep_caps = {
+    'ConfigureDigitalEdgeScriptTrigger':        { 'parameters': { 1: { 'is_repeated_capability': True, }, }, },
+    'ConfigureDigitalLevelScriptTrigger':       { 'parameters': { 1: { 'is_repeated_capability': True, }, }, },
+    'ConfigureSoftwareEdgeScriptTrigger':       { 'parameters': { 1: { 'is_repeated_capability': True, }, }, },
+    'DisableScriptTrigger':                     { 'parameters': { 1: { 'is_repeated_capability': True, }, }, },
+    'SendSoftwareEdgeTrigger':                  { 'parameters': { 2: { 'is_repeated_capability': True, }, }, },
```
Comment (texasaggie97) [inline src/nifgen/metadata/functions_addon.py]: Should `'SendSoftwareEdgeTrigger'` change this? Is it used differently?

Comment (marcoskirsch) [reply src/nifgen/metadata/functions_addon.py]: This is the C signature:

``` C
ViStatus niFgen_SendSoftwareEdgeTrigger (ViSession vi, viInt32 trigger, viConstString triggerID);
```

Looking at [documentation](http://zone.ni.com/reference/en-XX/help/370524T-01/fgencref/cvinifgen_sendsoftwareedgetrigger/), the `trigger` parameter should be an enum and the `triggerID` parameter should allow for string `"None"` as well. Unless the driver treats `""` and `VI_NULL` to mean the same as `"None"`.

File: src/nifgen/metadata/functions_addon.py [hunk 2]
```diff
@@ -84,8 +84,17 @@
...
+}
+
 functions_issues = {
-    'GetFIRFilterCoefficients':             { 'parameters': { 3: {'direction':'out'},  # TODO(marcoskirsch): Remove when #534 solved
+    'GetFIRFilterCoefficients':             { 'parameters': { 3: { 'direction':'out'},  # TODO(marcoskirsch): Remove when #534 solved
```
Comment (marcoskirsch) [inline src/nifgen/metadata/functions_addon.py]: #534 is closed so it appears the comment is stale.

File: src/nifgen/system_tests/test_system_nifgen.py [hunk 3]
```diff
@@ -389,14 +389,14 @@ def test_fir_filter_coefficients():
...
-    session.configure_digital_level_script_trigger('ScriptTrigger0', 'PXI_Trig0', nifgen.TriggerWhen.HIGH)
+    session.script_triggers[0].configure_digital_level_script_trigger('PXI_Trig0', nifgen.TriggerWhen.HIGH)
 
 
 def test_send_software_edge_trigger(session):
     waveform_data = [x * (1.0 / 256.0) for x in range(256)]
     session.create_waveform(waveform_data)
     with session.initiate():
-        session.send_software_edge_trigger(nifgen.Trigger.SCRIPT, 'ScriptTrigger0')
+        session.script_triggers[0].send_software_edge_trigger(nifgen.Trigger.SCRIPT)
```
Comment (marcoskirsch) [inline src/nifgen/system_tests/test_system_nifgen.py]: Gotta add a test case for "None".

File: src/nifgen/metadata/functions_addon.py (hunk 1 above)
Comment (texasaggie97) [reply src/nifgen/metadata/functions_addon.py]: There is a `Trigger` enum in enums_addon. Should I use that? It was waiting on #624 which is closed.

I don't think repeated capabilities as currently working will support "None". It will become "ScriptTriggerNone" which is not what we want for this particular use case.

Should maybe this function not use the repeated capability? It seems like a different use.

Comment (marcoskirsch) [reply src/nifgen/metadata/functions_addon.py]: > Should maybe this function not use the repeated capability? It seems like a different use.

I definitely don't think it works as it is in the PR right now:

``` python
send_software_edge_trigger(trigger)
```

Valid use cases would be:

``` python
session.script_triggers[1].send_software_edge_trigger(nifgen.Trigger.SCRIPT)
session["None"].send_software_edge_trigger(nifgen.Trigger.START)
```

If you use the `script_triggers container` then trigger **must be** `nifgen.Trigger.SCRIPT `. That's silly and redundant.

User has no good way of knowing `session.script_triggers["None"]` is what myst be passed if you want to send a start software trigger.

If we didn't make `viConstString triggerID` a repeated capability:

``` python
session.send_software_edge_trigger(nifgen.Trigger.SCRIPT, "ScriptTrigger1")
session.send_software_edge_trigger(nifgen.Trigger.START, "None")
```

User would need to know the magic strings. This is weird but we can live with it.

Ideally, arguably, usage should be:

``` python
session.script_triggers[1].send_software_edge_trigger()
session.send_software_edge_trigger() # sends start trigger
```

No enum needed. I'm not sure this is the best way to go, and I think it would require at the very least a fancy wrapper.

Do other APIs have the same signature for sending software triggers?

Comment (marcoskirsch) [reply src/nifgen/metadata/functions_addon.py]: For expediency, I suggest removing the repeated capability from this function and opening an issue to track it / decide on it separately.

File: CHANGELOG.md [hunk 4]
```diff
@@ -60,6 +60,9 @@ All notable changes to this project will be documented in this file.
 * ### NI-FGEN
     * #### Added
     * #### Changed
+        * Some functions missed setting repeated capabilities, leaving these as parameters instead of using the repeated capabilites object.
+            * `session.configure_digital_edge_script_trigger('0', source, ...)` becomes `session.script_triggers[0].configure_digital_edge_script_trigger(source, ...)`
+            * `session.configure_digital_level_script_trigger('0', source, ...)` becomes `session.script_triggers[0].configure_digital_level_script_trigger(source, ...)`
```
Comment (marcoskirsch) [inline CHANGELOG.md]: The "before" example is wrong. It should be  `session.configure_digital_level_script_trigger('ScriptTrigger0', source, ...)`

File: generated/nifgen/session.py [hunk 5]
```diff
@@ -1496,6 +1496,168 @@ def configure_custom_fir_filter_coefficients(self, coefficients_array):
...
+
+        Configures the specified Script Trigger for digital level triggering.
+
+        Tip:
+        This method requires repeated capabilities (usually channels). If called directly on the
+        nifgen.Session object, then the method will use all repeated capabilities in the session.
+        You can specify a subset of repeated capabilities using the Python index notation on an
+        nifgen.Session instance, and calling this method on the result.:
+
+            session.channels['0,1'].configure_digital_level_script_trigger(source, trigger_when)
```
Comment (marcoskirsch) [inline generated/nifgen/session.py]: #851 refers to this. :(

File: src/nifgen/metadata/functions_addon.py [hunk 6]
```diff
@@ -85,8 +85,17 @@
     'ConfigureDigitalLevelScriptTrigger':       { 'parameters': { 3: { 'enum': 'TriggerWhen',               }, }, },
 }
 
+# Need to manually set some repeated capabilities
+functions_rep_caps = {
+    'ConfigureDigitalEdgeScriptTrigger':        { 'parameters': { 1: { 'is_repeated_capability': True, }, }, },
```
Comment (marcoskirsch) [inline src/nifgen/metadata/functions_addon.py]: I want to confirm that we decided to keep these functions and also the corresponding properties?
In a review of `nidcpower` we had said to remove the duplicate functionality.

File: CHANGELOG.md (hunk 4 above)
Comment (texasaggie97) [reply CHANGELOG.md]: Fixed

File: docs/nifgen/functions.rst [hunk 7]
```diff
@@ -474,35 +474,23 @@ nifgen.Session methods
...
 
-.. py:method:: configure_digital_edge_script_trigger(trigger_id, source, edge=nifgen.ScriptTriggerDigitalEdgeEdge.RISING)
+.. py:method:: configure_digital_edge_script_trigger(source, edge=nifgen.ScriptTriggerDigitalEdgeEdge.RISING)
 
     Configures the specified Script Trigger for digital edge triggering.
 
     
 
 
+    .. tip:: This method requires repeated capabilities (usually channels). If called directly on the
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply docs/nifgen/functions.rst]: We will need to either add or infer more information than we do now. At this point, all we know is that there is a repeated capability used in the function, not which one. It is up to the use to know whether to use `.channel[]` or `.script_trigger[]`.

Because we don't have this additional information (yet?) we have no way to specialize the tip.

File: src/nifgen/metadata/functions_addon.py (hunk 6 above)
Comment (texasaggie97) [reply src/nifgen/metadata/functions_addon.py]: This PR may become moot once we review fgen. 

However, `nidcpower` doesn't have ScriptTriggers so the functions here may have more functionality other than just setting attributes.

File: src/nifgen/metadata/functions_addon.py (hunk 2 above)
Comment (texasaggie97) [reply src/nifgen/metadata/functions_addon.py]: Looking at #534, I am not sure it is really closed. Looking at the code that closed it, I am not sure how it was supposed to.

File: src/nifgen/system_tests/test_system_nifgen.py [hunk 8]
```diff
@@ -398,7 +398,7 @@ def test_fir_filter_coefficients():
 
 def test_configure_triggers(session):
     session.configure_digital_edge_start_trigger('PFI0', nifgen.StartTriggerDigitalEdgeEdge.FALLING)
-    session.configure_digital_level_script_trigger('ScriptTrigger0', 'PXI_Trig0', nifgen.TriggerWhen.HIGH)
+    session.script_triggers[0].configure_digital_level_script_trigger('PXI_Trig0', nifgen.TriggerWhen.HIGH)
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply src/nifgen/system_tests/test_system_nifgen.py]: `configure_software_edge_script_trigger()` and `disable_script_trigger()` are not called in the system test, therefore there is nothing to change.

[the remark this answers is not in the document]
Comment (texasaggie97) [reply src/nifgen/system_tests/test_system_nifgen.py]: These methods are not in the public API, so I have removed both `configure_software_edge_script_trigger()` and `disable_script_trigger()` from function_addons.py
